src/reducers/linter.tsx

```diff
@@ -177,7 +177,12 @@ export const fetchLinterMessages = ({
   url: string;
   versionId: number;
 }): ThunkActionCreator => {
-  return async (dispatch) => {
+  return async (dispatch, getState) => {
+    // See: https://github.com/mozilla/addons-code-manager/issues/591
+    if (getState().linter.isLoading) {
```

Comment:
Please remove the `isLoading` logic from [LinterProvider](https://github.com/mozilla/addons-code-manager/blob/50be0327f1ae670d743a570855d23bef3eff14c3/src/components/LinterProvider/index.tsx#L75) (and tests) because we no longer need it.

Reply:
I thought about doing this but the existing logic avoids a `dispatch` call on update, so that still seems worthy, no?

Reply:
Answering myself: controlling the `isLoading` logic in one location is actually better.